src/App.tsx

```diff
@@ -1,14 +1,71 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
+import { getTodos, getUser } from './api';
+import { Todo } from './types/Todo';
+import { FilterStatus } from './types/FilterStatus';
 import { TodoModal } from './components/TodoModal';
+import { User } from './types/User';
 import { Loader } from './components/Loader';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [filteredTodos, setFilteredTodos] = useState<Todo[]>([]);
+  const [selectedTodo, setSelectedTodo] = useState<Todo | null>(null);
+  const [currentUser, setCurrentUser] = useState<User | null>(null);
+  const [todosLoading, setTodoLoading] = useState(true);
+
+  useEffect(() => {
+    getTodos()
+      .then(fetchedTodos => {
+        setTodos(fetchedTodos);
+        setFilteredTodos(fetchedTodos);
+      })
+      .finally(() => {
+        setTodoLoading(false);
+      });
+  }, []);
+
+  const filterList = (status: FilterStatus, query: string) => {
+    const preparedTodos = todos.filter((todo: Todo) => {
+      const matchesStatus =
+        status === 'all' ||
+        (status === 'active' && !todo.completed) ||
+        (status === 'completed' && todo.completed);
+
+      const matchesTitle = todo.title
+        .trim()
+        .toLowerCase()
+        .includes(query.trim().toLowerCase());
+
+      return matchesStatus && matchesTitle;
+    });
+
+    return preparedTodos;
+  };
+
+  const handleFilterChange = (status: FilterStatus, query: string) => {
+    const newFilteredTodos = filterList(status, query);
+
+    setFilteredTodos(newFilteredTodos);
+  };
+
+  const handleTodoSelection = (todo: Todo) => {
+    setSelectedTodo(todo);
+    getUser(todo.userId).then(user => {
+      setCurrentUser(user);
+    });
```

Comment:
The `getUser` function call in `handleTodoSelection` does not handle errors. Consider adding a `.catch()` block to handle any potential errors when fetching user data. This will ensure that your application can handle scenarios where the user data cannot be retrieved.